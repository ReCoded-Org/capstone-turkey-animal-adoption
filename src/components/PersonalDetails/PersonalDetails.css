.newColor {
  color: #ff5555;
}

h3.newColor,
h4.newColor,
h5.newColor,
label.newColor {
  font-weight: bold;
}

p,
.email {
  color: #777777;
}

.email {
  font-size: 13px;
  margin-top: -2px;
}

.avatar {
  border-radius: 50%;
  background: #999;
}

.inputStyle {
  border: 0.5px solid;
  color: #444444;
  border: 1px solid #ff5555;
  width: 190px;
  height: 30px;
}

.inputStyle:disabled {
  background-color: #e9ecef;
  opacity: 1;
}

.inputStyle::placeholder {
  color: #777777;
}

textarea.inputStyle {
  height: 50px;
}

.button {
  background: #3f6487;
  color: white;
  width: 200px;
  height: 40px;
  border-radius: 40px;
  border: 0;
}

.cancelBtn {
  background: #777777;
}

.mainContainer {
  background-color: #e5e5e5;
}

.bgWhite {
  background-color: white;
}

.fullHeight {
  height: 100%;
}

.profile .error {
  color: #ff5555;
  font-size: 12px;
  margin-top: 2px;
}
