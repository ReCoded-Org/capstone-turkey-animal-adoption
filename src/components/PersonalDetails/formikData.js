import * as Yup from "yup";
const initialValues = {
  fullname: "",
  email: "",
  phone: "",
  bio: "",
  city: "",
  state: "",
  street: "",
  zip: "",
};

const yupValidation = {
  phone: Yup.number().required("Phone is Required"),
  bio: Yup.string().required("Bio is Required"),
  city: Yup.string().required("City is Required"),
  state: Yup.string().required("State is Required"),
  street: Yup.string().required("Street is Required"),
};

export { initialValues, yupValidation };
