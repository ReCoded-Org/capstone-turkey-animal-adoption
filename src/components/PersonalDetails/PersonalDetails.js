import React, { useState } from "react";
import { Col, Row, Container, Alert } from "react-bootstrap";
import Person from "./person.png";
import { Formik, Field, Form, ErrorMessage } from "formik";
import { initialValues, yupValidation } from "./formikData";
import * as Yup from "yup";
import cities from "../../data/cities";
import "./PersonalDetails.css";
import { useTranslation } from "react-i18next";

const PersonalDetails = () => {
  const [bio, setBio] = useState(null);
  const [alertShown, isAlertShown] = useState(false);

  const { t } = useTranslation();

  return (
    <Container fluid className="mainContainer profile">
      <Container className="pt-3 pb-4">
        <Row>
          <Col lg="4" md="12">
            <div class="p-5 shadow-lg bgWhite fullHeight">
              <img
                src={Person}
                alt="userPhoto"
                width="120"
                height="120"
                className="align-items-center mb-4 avatar"
              />
              <h5 className="align-items-center mb-2 text-uppercase newColor">
                Mohammed Manar Yazji
              </h5>
              <p className="align-items-center  email">
                yazjiamanr92@gmail.com
              </p>
              {bio && (
                <h4 className="align-items-center mb-2 mt-4 newColor">About</h4>
              )}
              <p className="text-center">{bio}</p>
            </div>
          </Col>
          <Col lg="8" md="12" className="mt-lg-0 mt-5">
            <div className="p-5 shadow-lg bgWhite" align="left">
              {alertShown && (
                <Alert variant="success">Your profile has been updated</Alert>
              )}
              <Formik
                initialValues={initialValues}
                validationSchema={Yup.object(yupValidation)}
                onSubmit={(values, { setSubmitting }) => {
                  isAlertShown(true);
                }}
              >
                {(props) => {
                  const { values } = props;
                  return (
                    <Form>
                      {setBio(values.bio)}
                      <h3 className="newColor">{t("personalPage.title0")}</h3>

                      <div className="row d-flex justify-content-around mt-4 ">
                        <div>
                          <label className="newColor">
                            {t("personalPage.subTitle0")}
                          </label>
                          <br />
                          <Field
                            placeholder={t("personalPage.placeholder0")}
                            type="text"
                            name="fullname"
                            className="shadow rounded inputStyle"
                          />
                          <div className="error">
                            <ErrorMessage name="fullname" />
                          </div>
                        </div>
                        <div>
                          <label className="newColor">
                            {t("personalPage.subTitle1")}
                          </label>
                          <br />
                          <input
                            placeholder={t("personalPage.placeholder1")}
                            type="Email"
                            name="email"
                            value="yazjiamanr92@gmail.com"
                            disabled
                            className="shadow rounded inputStyle"
                          />
                        </div>
                      </div>
                      <div className="row d-flex justify-content-around mt-4">
                        <div>
                          <label className="newColor">
                            {t("personalPage.subTitle2")}
                          </label>
                          <br />
                          <Field
                            placeholder={t("personalPage.placeholder2")}
                            type="Number"
                            name="phone"
                            className="shadow rounded inputStyle"
                          />
                          <div className="error">
                            <ErrorMessage name="phone" />
                          </div>
                        </div>
                        <div>
                          <label className="newColor">
                            {t("personalPage.subTitle3")}
                          </label>
                          <br />
                          <Field
                            placeholder={t("personalPage.placeholder3")}
                            name="bio"
                            as="textarea"
                            className="shadow rounded inputStyle"
                          />
                          <div className="error">
                            <ErrorMessage name="bio" />
                          </div>
                        </div>
                      </div>
                      <div className="mt-4">
                        <h3 className="newColor">{t("personalPage.title1")}</h3>
                        <div className="row d-flex justify-content-around mt-4 ">
                          <div>
                            <label className="newColor">
                              {t("personalPage.subTitle4")}
                            </label>
                            <br />
                            <Field
                              component="select"
                              id="city"
                              name="city"
                              multiple={false}
                              className="shadow rounded inputStyle"
                            >
                              <option value="">
                                {t("personalPage.dropdown")}
                              </option>
                              {cities.map((city) => {
                                return (
                                  <option value="city.name">{city.name}</option>
                                );
                              })}
                            </Field>
                            <div className="error">
                              <ErrorMessage name="city" />
                            </div>
                          </div>
                          <div>
                            <label className="newColor">
                              {t("personalPage.subTitle5")}
                            </label>
                            <br />
                            <Field
                              placeholder={t("personalPage.placeholder5")}
                              type="text"
                              name="state"
                              className="shadow rounded inputStyle"
                            />
                            <div className="error">
                              <ErrorMessage name="state" />
                            </div>
                          </div>
                        </div>
                        <div className="row d-flex justify-content-around mt-4 ">
                          <div>
                            <label className="newColor">
                              {t("personalPage.subTitle6")}
                            </label>
                            <br />
                            <Field
                              placeholder={t("personalPage.placeholder6")}
                              type="text"
                              name="street"
                              className="shadow rounded inputStyle"
                            />
                            <div className="error">
                              <ErrorMessage name="street" />
                            </div>
                          </div>
                          <div>
                            <label className="newColor">
                              {t("personalPage.subTitle7")}
                            </label>
                            <br />
                            <Field
                              placeholder={t("personalPage.placeholder7")}
                              type="text"
                              name="zip"
                              className="shadow rounded inputStyle"
                            />
                            <div className="error">
                              <ErrorMessage name="zip" />
                            </div>
                          </div>
                        </div>
                        <div className="row d-flex mt-5 justify-content-center">
                          <input
                            className="m-2 button"
                            type="submit"
                            value="Update"
                          />
                          <button className="m-2 button cancelBtn">
                            {t("personalPage.button")}{" "}
                          </button>
                        </div>
                      </div>
                    </Form>
                  );
                }}
              </Formik>
            </div>
          </Col>
        </Row>
      </Container>
    </Container>
  );
};

export default PersonalDetails;
