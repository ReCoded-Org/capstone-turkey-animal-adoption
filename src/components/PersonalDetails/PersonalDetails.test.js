import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import PersonalDetails from "./PersonalDetails";

it("Persnal details renders correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <PersonalDetails />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
