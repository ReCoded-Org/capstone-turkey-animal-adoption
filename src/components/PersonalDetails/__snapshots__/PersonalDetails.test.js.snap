// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Persnal details renders correctly 1`] = `
<div
  className="mainContainer profile container-fluid"
>
  <div
    className="pt-3 pb-4 container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4 col-md-12"
      >
        <div
          class="p-5 shadow-lg bgWhite fullHeight"
        >
          <img
            alt="userPhoto"
            className="align-items-center mb-4 avatar"
            height="120"
            src="person.png"
            width="120"
          />
          <h5
            className="align-items-center mb-2 text-uppercase newColor"
          >
            Mohammed Manar Yazji
          </h5>
          <p
            className="align-items-center  email"
          >
            yazjiamanr92@gmail.com
          </p>
          
          <p
            className="text-center"
          >
            
          </p>
        </div>
      </div>
      <div
        className="mt-lg-0 mt-5 col-lg-8 col-md-12"
      >
        <div
          align="left"
          className="p-5 shadow-lg bgWhite"
        >
          <form
            action="#"
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <h3
              className="newColor"
            >
              Personal Details
            </h3>
            <div
              className="row d-flex justify-content-around mt-4 "
            >
              <div>
                <label
                  className="newColor"
                >
                  FullName
                </label>
                <br />
                <input
                  className="shadow rounded inputStyle"
                  name="fullname"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Full Name"
                  type="text"
                  value=""
                />
                <div
                  className="error"
                />
              </div>
              <div>
                <label
                  className="newColor"
                >
                  Email
                </label>
                <br />
                <input
                  className="shadow rounded inputStyle"
                  disabled={true}
                  name="email"
                  placeholder="Email"
                  type="Email"
                  value="yazjiamanr92@gmail.com"
                />
              </div>
            </div>
            <div
              className="row d-flex justify-content-around mt-4"
            >
              <div>
                <label
                  className="newColor"
                >
                  Phone
                </label>
                <br />
                <input
                  className="shadow rounded inputStyle"
                  name="phone"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Phone Number"
                  type="Number"
                  value=""
                />
                <div
                  className="error"
                />
              </div>
              <div>
                <label
                  className="newColor"
                >
                  Bio
                </label>
                <br />
                <textarea
                  className="shadow rounded inputStyle"
                  name="bio"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Bio"
                  value=""
                />
                <div
                  className="error"
                />
              </div>
            </div>
            <div
              className="mt-4"
            >
              <h3
                className="newColor"
              >
                Address
              </h3>
              <div
                className="row d-flex justify-content-around mt-4 "
              >
                <div>
                  <label
                    className="newColor"
                  >
                    City
                  </label>
                  <br />
                  <select
                    className="shadow rounded inputStyle"
                    id="city"
                    multiple={false}
                    name="city"
                    onBlur={[Function]}
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value=""
                    >
                      Select a City
                    </option>
                    <option
                      value="city.name"
                    >
                      adana
                    </option>
                    <option
                      value="city.name"
                    >
                      adıyaman
                    </option>
                    <option
                      value="city.name"
                    >
                      afyonkarahisar
                    </option>
                    <option
                      value="city.name"
                    >
                      ağrı
                    </option>
                    <option
                      value="city.name"
                    >
                      amasya
                    </option>
                    <option
                      value="city.name"
                    >
                      ankara
                    </option>
                    <option
                      value="city.name"
                    >
                      antalya
                    </option>
                    <option
                      value="city.name"
                    >
                      artvin
                    </option>
                    <option
                      value="city.name"
                    >
                      aydın
                    </option>
                    <option
                      value="city.name"
                    >
                      balıkesir
                    </option>
                    <option
                      value="city.name"
                    >
                      bilecik
                    </option>
                    <option
                      value="city.name"
                    >
                      bingöl
                    </option>
                    <option
                      value="city.name"
                    >
                      bitlis
                    </option>
                    <option
                      value="city.name"
                    >
                      bolu
                    </option>
                    <option
                      value="city.name"
                    >
                      burdur
                    </option>
                    <option
                      value="city.name"
                    >
                      bursa
                    </option>
                    <option
                      value="city.name"
                    >
                      çanakkale
                    </option>
                    <option
                      value="city.name"
                    >
                      çankırı
                    </option>
                    <option
                      value="city.name"
                    >
                      çorum
                    </option>
                    <option
                      value="city.name"
                    >
                      denizli
                    </option>
                    <option
                      value="city.name"
                    >
                      diyarbakır
                    </option>
                    <option
                      value="city.name"
                    >
                      edirne
                    </option>
                    <option
                      value="city.name"
                    >
                      elazığ
                    </option>
                    <option
                      value="city.name"
                    >
                      erzincan
                    </option>
                    <option
                      value="city.name"
                    >
                      erzurum
                    </option>
                    <option
                      value="city.name"
                    >
                      eskişehir
                    </option>
                    <option
                      value="city.name"
                    >
                      gaziantep
                    </option>
                    <option
                      value="city.name"
                    >
                      giresun
                    </option>
                    <option
                      value="city.name"
                    >
                      gümüşhane
                    </option>
                    <option
                      value="city.name"
                    >
                      hakkari
                    </option>
                    <option
                      value="city.name"
                    >
                      hatay
                    </option>
                    <option
                      value="city.name"
                    >
                      ısparta
                    </option>
                    <option
                      value="city.name"
                    >
                      mersin
                    </option>
                    <option
                      value="city.name"
                    >
                      istanbul
                    </option>
                    <option
                      value="city.name"
                    >
                      izmir
                    </option>
                    <option
                      value="city.name"
                    >
                      kars
                    </option>
                    <option
                      value="city.name"
                    >
                      kastamonu
                    </option>
                    <option
                      value="city.name"
                    >
                      kayseri
                    </option>
                    <option
                      value="city.name"
                    >
                      kırklareli
                    </option>
                    <option
                      value="city.name"
                    >
                      kırşehir
                    </option>
                    <option
                      value="city.name"
                    >
                      kocaeli
                    </option>
                    <option
                      value="city.name"
                    >
                      konya
                    </option>
                    <option
                      value="city.name"
                    >
                      kütahya
                    </option>
                    <option
                      value="city.name"
                    >
                      malatya
                    </option>
                    <option
                      value="city.name"
                    >
                      manisa
                    </option>
                    <option
                      value="city.name"
                    >
                      kahramanmaraş
                    </option>
                    <option
                      value="city.name"
                    >
                      mardin
                    </option>
                    <option
                      value="city.name"
                    >
                      muğla
                    </option>
                    <option
                      value="city.name"
                    >
                      muş
                    </option>
                    <option
                      value="city.name"
                    >
                      nevşehir
                    </option>
                    <option
                      value="city.name"
                    >
                      niğde
                    </option>
                    <option
                      value="city.name"
                    >
                      ordu
                    </option>
                    <option
                      value="city.name"
                    >
                      rize
                    </option>
                    <option
                      value="city.name"
                    >
                      sakarya
                    </option>
                    <option
                      value="city.name"
                    >
                      samsun
                    </option>
                    <option
                      value="city.name"
                    >
                      siirt
                    </option>
                    <option
                      value="city.name"
                    >
                      sinop
                    </option>
                    <option
                      value="city.name"
                    >
                      sivas
                    </option>
                    <option
                      value="city.name"
                    >
                      tekirdağ
                    </option>
                    <option
                      value="city.name"
                    >
                      tokat
                    </option>
                    <option
                      value="city.name"
                    >
                      trabzon
                    </option>
                    <option
                      value="city.name"
                    >
                      tunceli
                    </option>
                    <option
                      value="city.name"
                    >
                      şanlıurfa
                    </option>
                    <option
                      value="city.name"
                    >
                      uşak
                    </option>
                    <option
                      value="city.name"
                    >
                      van
                    </option>
                    <option
                      value="city.name"
                    >
                      yozgat
                    </option>
                    <option
                      value="city.name"
                    >
                      zonguldak
                    </option>
                    <option
                      value="city.name"
                    >
                      aksaray
                    </option>
                    <option
                      value="city.name"
                    >
                      bayburt
                    </option>
                    <option
                      value="city.name"
                    >
                      karaman
                    </option>
                    <option
                      value="city.name"
                    >
                      kırıkkale
                    </option>
                    <option
                      value="city.name"
                    >
                      batman
                    </option>
                    <option
                      value="city.name"
                    >
                      şırnak
                    </option>
                    <option
                      value="city.name"
                    >
                      bartın
                    </option>
                    <option
                      value="city.name"
                    >
                      ardahan
                    </option>
                    <option
                      value="city.name"
                    >
                      ığdır
                    </option>
                    <option
                      value="city.name"
                    >
                      yalova
                    </option>
                    <option
                      value="city.name"
                    >
                      karabük
                    </option>
                    <option
                      value="city.name"
                    >
                      kilis
                    </option>
                    <option
                      value="city.name"
                    >
                      osmaniye
                    </option>
                    <option
                      value="city.name"
                    >
                      düzce
                    </option>
                  </select>
                  <div
                    className="error"
                  />
                </div>
                <div>
                  <label
                    className="newColor"
                  >
                    State
                  </label>
                  <br />
                  <input
                    className="shadow rounded inputStyle"
                    name="state"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="State Name"
                    type="text"
                    value=""
                  />
                  <div
                    className="error"
                  />
                </div>
              </div>
              <div
                className="row d-flex justify-content-around mt-4 "
              >
                <div>
                  <label
                    className="newColor"
                  >
                    Street
                  </label>
                  <br />
                  <input
                    className="shadow rounded inputStyle"
                    name="street"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Street"
                    type="text"
                    value=""
                  />
                  <div
                    className="error"
                  />
                </div>
                <div>
                  <label
                    className="newColor"
                  >
                    Zip Code
                  </label>
                  <br />
                  <input
                    className="shadow rounded inputStyle"
                    name="zip"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Zip Code"
                    type="text"
                    value=""
                  />
                  <div
                    className="error"
                  />
                </div>
              </div>
              <div
                className="row d-flex mt-5 justify-content-center"
              >
                <input
                  className="m-2 button"
                  type="submit"
                  value="Update"
                />
                <button
                  className="m-2 button cancelBtn"
                >
                  Cancel
                   
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
